src/plugins/iam/services/TwitterAppSubscriber.js

```diff
@@ -0,0 +1,136 @@
+    super();
+    this.onInitForm = this.onInitForm.bind(this);
+    this.onWarnForm = this.onWarnForm.bind(this);
+    this.onValidateForm = this.onValidateForm.bind(this);
+    this.onSubmitForm = this.onSubmitForm.bind(this);
+  }
+
+  /**
+   * Runs when the form is first created and is used to
+   * update initial values.
+   *
+   * @param {FormEvent} formEvent
+   *
+   */
+  onInitForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    [
+      'oauth_consumer_key',
+      'oauth_consumer_secret',
+      'oauth_token',
+      'oauth_token_secret',
+    ].forEach((schemaFieldName) => {
+      const formFieldName = camelCase(schemaFieldName);
+      data[formFieldName] = node.get(schemaFieldName);
+    });
+  }
+
+  /**
+   * Occurs before validation and is used to add warnings.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onWarnForm(formEvent) {
+    const data = formEvent.getData();
+    const { oauthConsumerSecret, oauthTokenSecret } = data;
+
+    // TODO: fix warning, it's not working here or in apple news subscriber
+
+    if (!oauthConsumerSecret && !oauthTokenSecret) {
+      return;
+    }
+
+    if (oauthConsumerSecret && oauthConsumerSecret.length <= 50) {
+      formEvent.addWarning(
+        'oauthConsumerSecret',
+        'Oauth consumer secret should be longer than what currently entered. Make sure the oauth consumer secret is encrypted',
+      );
+    }
+
+    if (oauthTokenSecret && oauthTokenSecret.length <= 45) {
+      formEvent.addWarning(
+        'oauthTokenSecret',
+        'Oauth token secret should be longer than what currently entered. Make sure the oauth token secret is encrypted',
+      );
+    }
+  }
+
+  /**
+   * This is the redux form sync validation process.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onValidateForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    if (!formEvent.getProps().isCreateForm) {
+      let error;
+      // Fields using TextField component
+      ['oauth_consumer_key', 'oauth_token'].forEach((fieldName) => {
+        error = getTextFieldError(data, camelCase(fieldName), node);
+        if (error) {
+          formEvent.addError(camelCase(fieldName), error);
+        }
+      });
+
+      ['oauth_consumer_secret', 'oauth_token_secret'].forEach((fieldName) => {
+        error = getTextAreaFieldError(data, camelCase(fieldName), node);
+        if (error) {
+          formEvent.addError(camelCase(fieldName), error);
+        }
+      });
+    }
+  }
+
+  /**
+   * Binds data from the redux form to the command.
+   * This occurs AFTER a form has been submitted
+   * but before the command is dispatched.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onSubmitForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    if (node.isFrozen()) {
+      return;
+    }
+
+    if (!formEvent.getProps().isCreateForm) {
```

Comment:
@aloncohenn bit cleaner to do an early return here

Reply:
@efjacobson this is consistent with how all the other ones do it, do we want to make this one different or keep the same?

Reply:
@aloncohenn hmm..dont think there is a reason they have to be like that. anyways not worth it, leave as is